Switch CI for stable julia from 1.11 to 1.12

visr opened the pull request:
Some changes for 1.12 have already been merged, see https://github.com/JuliaLang/PackageCompiler.jl/pull/1047 and https://github.com/JuliaLang/PackageCompiler.jl/pull/1068.

This contains several test and code fixes to get the tests to pass on Julia 1.12.4.

https://github.com/JuliaLang/PackageCompiler.jl/pull/1078/changes/36460939271a473c51ed6b68745298b3b84291e7 fixes https://github.com/JuliaLang/PackageCompiler.jl/issues/1084

visr committed:
Switch CI for stable julia from 1.11 to 1.12

Some changes for 1.12 have already been merged, see https://github.com/JuliaLang/PackageCompiler.jl/pull/1047 and https://github.com/JuliaLang/PackageCompiler.jl/pull/1068.

It would be good to run CI to confirm this is enough to pass tests.

KristofferC commented:
https://github.com/JuliaLang/PackageCompiler.jl/issues/1076 is still an issue. https://github.com/JuliaLang/julia/pull/60383 should fix that (but requires a new julia release).

visr committed:
Fix nthreads test

visr committed:
Exclude MbedTLS from sysimage stdlibs in 1.12+

visr committed:
Split build-mylib manifests

CI was passing on julia 1.10, but failing on julia 1.12 with `ERROR: LoadError: Failed to find path for package MbedTLS_jll`. I renamed the existing Manifest.toml to Manifest-v1.10.toml, and ran `upgrade_manifest` and `resolve` on Manifest.toml to ensure it supports julia 1.12+.

visr committed:
Try to fix CI error:

```
2026-01-12T16:14:10.4616725Z gcc my_application.c -o my_application.out -I"MyLibCompiled"/include -L"MyLibCompiled"/lib -ljulia -lmylib
2026-01-12T16:14:10.5564717Z ##[group]Run ./examples/MyLib/my_application.out
2026-01-12T16:14:10.5565064Z [36;1m./examples/MyLib/my_application.out[0m
2026-01-12T16:14:10.5593873Z shell: /usr/bin/bash -e {0}
2026-01-12T16:14:10.5594105Z env:
2026-01-12T16:14:10.5594311Z   JULIA_PKG_SERVER_REGISTRY_PREFERENCE: eager
2026-01-12T16:14:10.5594620Z   LD_LIBRARY_PATH: examples/MyLib/MyLibCompiled/lib
2026-01-12T16:14:10.5594882Z ##[endgroup]
2026-01-12T16:14:10.5720640Z ERROR: could not load library "/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/examples/MyLib/MyLibCompiled/lib/../bin/examples/MyLib/MyLibCompiled/lib/libmylib.so"
2026-01-12T16:14:10.5722843Z /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/examples/MyLib/MyLibCompiled/lib/../bin/examples/MyLib/MyLibCompiled/lib/libmylib.so: cannot open shared object file: No such file or directory
```

visr commented on examples/MyLib/build/Manifest-v1.10.toml:
CI was passing on julia 1.10, but failing on julia 1.12 with `ERROR: LoadError: Failed to find path for package MbedTLS_jll`. I renamed the existing Manifest.toml to Manifest-v1.10.toml, and ran `upgrade_manifest` and `resolve` on Manifest.toml to ensure it supports julia 1.12+.

visr commented on src/julia_init.c:
@gbaraldi does this change in `julia_init.c` make sense to you? I'm no C coder and used AI assistance here, but given  https://github.com/JuliaLang/PackageCompiler.jl/pull/1047 and https://github.com/JuliaLang/julia/pull/58231 you can probably tell. This does fix build-mylib CI on 1.12, which was failing with

```
ERROR: could not load library "/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/examples/MyLib/MyLibCompiled/lib/../bin/examples/MyLib/MyLibCompiled/lib/libmylib.so"
```

That path was clearly incorrect, so this switches from

```c
jl_init_with_image_file(NULL, sysimage_path)
```
to
```c
jl_init_with_image_file(bindir, abs_sysimage_path);
```
to get the right path.

visr reviewed (commented):
CI looks green, would appreciate a review.

The Windows 32 bit tests are failing but that is a pre-existing issue: https://github.com/JuliaLang/PackageCompiler.jl/issues/1083.

visr committed:
Compile fresh sysimage with one thread to avoid segfault on Julia 1.12

Fixes https://github.com/JuliaLang/PackageCompiler.jl/issues/1084
This is only a problem in JULIA_NUM_THREADS is set to more than 1.

(cherry picked from commit 870d0152e85abee93f661e902510acfb24f85373)

visr commented:
bump, would be great to have a review

KristofferC commented:
Thanks
